File: examples/typescript/comment-good.ts
// sum adds two numbers and returns the result.
function sum(a: number, b: number): number {
    return a + b;
}

// retry calls fn up to attempts times, waiting delay between failures.
// Returns the first successful result, or the last error if every attempt
// fails — callers that need cancellation should wrap fn themselves, since
// retry does not accept an AbortSignal.
function retry(attempts: number, delay: number, fn: () => number): number {
    let lastResult = 0;
    for (let i = 0; i < attempts; i++) {
        lastResult = fn();
    }
    return lastResult;
}

File: examples/python/comment-good.py
# sum_two adds two numbers and returns the result.
def sum_two(a, b):
    return a + b


# retry calls fn up to attempts times, waiting delay between failures.
# Returns the first successful result, or the last error if every attempt
# fails — callers that need cancellation should wrap fn themselves, since
# retry does not accept a timeout.
def retry(attempts, delay, fn):
    last_result = 0
    for _ in range(attempts):
        last_result = fn()
    return last_result
